src/main/java/team05/integrated_feed_backend/module/post/service/PostService.java

```diff
@@ -0,0 +1,38 @@
+package team05.integrated_feed_backend.module.post.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import lombok.RequiredArgsConstructor;
+import team05.integrated_feed_backend.common.code.StatusCode;
+import team05.integrated_feed_backend.exception.custom.BusinessException;
+import team05.integrated_feed_backend.module.post.dto.response.PostDetailRes;
+import team05.integrated_feed_backend.module.post.entity.Post;
+import team05.integrated_feed_backend.module.post.repository.PostRepository;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class PostService {
+
+	private final PostRepository postRepository;
+
+	@Transactional
+	public PostDetailRes getPostDetail(Long id) {
+
+		if (id == null || id <= 0) {
+			throw new BusinessException(StatusCode.BAD_REQUEST, StatusCode.BAD_REQUEST.getMessage());
+		}
+
+		Post post = postRepository.findDetailPostById(id);
+
+		if (post == null) {
+			throw new BusinessException(StatusCode.NOT_FOUND, StatusCode.NOT_FOUND.getMessage());
+		}
```

Comment:
💭 `findDetailPostById()`가 Optional을 반환하도록 변경한 다음, `orElseThrow()`를 사용해 보시는 건 어떠신가요?
```suggestion
		Post post = postRepository.findDetailPostById(id)
			.orElseThrow(() -> new BusinessException(StatusCode.NOT_FOUND));
```

Reply:
```suggestion
		Post post = postRepository.findDetailPostById(id)
			.orElseThrow(() -> new BusinessException(StatusCode.NOT_FOUND, StatusCode.NOT_FOUND.getMessage()));
```

감사합니다 :) 훨씬 깔끔하네용